ui/app/AppLayouts/Profile/views/profile/MyProfileSettingsView.qml

```diff
@@ -0,0 +1,152 @@
+import QtQuick 2.13
+import QtQuick.Controls 2.3
+import QtQuick.Layouts 1.13
+
+import utils 1.0
+import shared 1.0
+import shared.panels 1.0
+import shared.popups 1.0
+import shared.controls.chat 1.0
+
+import "../../popups"
+import "../../stores"
+import "../../controls"
+import "../../../Onboarding/shared" as OnboardingComponents
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components 0.1
+import StatusQ.Controls 0.1
+
+ColumnLayout {
+    id: root
+
+    property PrivacyStore privacyStore
+    property ProfileStore profileStore
+    property WalletStore walletStore
+
+    readonly property bool dirty: displayNameInput.text != profileStore.displayName
+                               || biometricsSwitch.checked != biometricsSwitch.currentStoredValue
+
+    readonly property bool valid: !!displayNameInput.text && displayNameInput.valid
+
+    function reset() {
+        displayNameInput.text = Qt.binding(() => { return profileStore.displayName })
+        biometricsSwitch.checked = Qt.binding(() => { return biometricsSwitch.currentStoredValue })
+    }
+
+    function save() {
+        profileStore.setDisplayName(displayNameInput.text)
+
+        if (biometricsSwitch.checked)
+            Global.openPopup(storePasswordModal)
+        else
+            localAccountSettings.storeToKeychainValue = Constants.storeToKeychainValueNever;
+    }
+
+    function offerToStorePassword(password, runStoreToKeyChainPopup)
+    {
+        if (Qt.platform.os !== "osx")
+            return;
+
+        localAccountSettings.storeToKeychainValue = Constants.storeToKeychainValueStore;
+        root.privacyStore.storeToKeyChain(password);
+    }
+
+    ProfileHeader {
+        Layout.fillWidth: true
+        Layout.leftMargin: Style.current.padding
+        Layout.rightMargin: Style.current.padding
+
+        displayName: profileStore.name
+        pubkey: profileStore.pubkey
+        icon: profileStore.icon
+        imageSize: ProfileHeader.ImageSize.Big
+        
+        displayNameVisible: false
+        pubkeyVisible: false
+        emojiHashVisible: false
+        editImageButtonVisible: true
+    }
+
+    StatusInput {
+        id: displayNameInput
+        Layout.fillWidth: true
+        label: qsTr("Display name")
+        input.placeholderText: qsTr("Display Name")
+        charLimit: 24
+        input.text: root.profileStore.displayName
+        validators: Constants.validators.displayName
+    }
+
+    StatusListItem {
+        Layout.fillWidth: true
+        visible: Qt.platform.os == "osx"
+        title: qsTr("Biometric login and transaction authentication")
+        icon.name: "touch-id"
+        components: [ StatusSwitch {
+            id: biometricsSwitch
+            readonly property bool currentStoredValue: localAccountSettings.storeToKeychainValue === Constants.storeToKeychainValueStore
```

Comment:
maybe I missed something, but what a reason to make new property just for one usage?